### PR TITLE
clementine: unstable-2022-04-11 -> 1.4.rc2-unstable-2024-05-12

### PR DESCRIPTION
**Pull request by pinpox**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/audio/clementine/default.nix [hunk 1]
```diff
@@ -50,12 +50,12 @@ let
   withCloud = config.clementine.cloud or true;
 in mkDerivation {
   pname = "clementine";
-  version = "unstable-2022-04-11";
+  version = "unstable-2023-10-02";
```
Comment (eclairevoyant) [inline pkgs/applications/audio/clementine/default.nix:53]:
```suggestion
  version = "1.3.1-unstable-2023-10-02";
```

File: pkgs/applications/audio/clementine/default.nix [hunk 2]
```diff
@@ -50,12 +50,12 @@ let
...
 
   src = fetchFromGitHub {
     owner = "clementine-player";
     repo = "Clementine";
-    rev = "250024e117fbe5fae7c62b9c8e655d66412a6ed7";
+    rev = "7b678f26e07faeeea7c3fc61d147c9693e29aa32";
```
Comment (eclairevoyant) [inline pkgs/applications/audio/clementine/default.nix:58]: Why this commit?

Comment (pinpox) [reply pkgs/applications/audio/clementine/default.nix:58]: It is the latest commit unless I did something wrong. Just bumping to the newest possible, as the package is running a non-release version anyway and the current commit is pretty outdated

Comment (eclairevoyant) [reply pkgs/applications/audio/clementine/default.nix:58]: latest commit is from last week: https://github.com/clementine-player/Clementine

File: pkgs/applications/audio/clementine/default.nix [hunk 3]
```diff
@@ -50,12 +50,12 @@ let
...
   pname = "clementine";
-  version = "unstable-2022-04-11";
+  version = "1.3.1-unstable-2023-10-02";
 
   src = fetchFromGitHub {
     owner = "clementine-player";
     repo = "Clementine";
-    rev = "250024e117fbe5fae7c62b9c8e655d66412a6ed7";
+    rev = "7b678f26e07faeeea7c3fc61d147c9693e29aa32";
     sha256 = "06fcbs3wig3mh711iypyj49qm5246f7qhvgvv8brqfrd8cqyh6qf";
```
Comment (eclairevoyant) [inline pkgs/applications/audio/clementine/default.nix:59]: Also, this hash needs to change, or you're just building the old rev

Comment (pinpox) [reply pkgs/applications/audio/clementine/default.nix]: Sorry, not sure how I missed that :see_no_evil:

**Comment by pinpox** [comment]
Added git as dependency and set an env var required to make the build succeed

File: pkgs/applications/audio/clementine/default.nix [hunk 4]
```diff
@@ -48,20 +49,22 @@ let
...
-    rev = "250024e117fbe5fae7c62b9c8e655d66412a6ed7";
-    sha256 = "06fcbs3wig3mh711iypyj49qm5246f7qhvgvv8brqfrd8cqyh6qf";
+    rev = "7b678f26e07faeeea7c3fc61d147c9693e29aa32";
+    sha256 = "sha256-bId2CxtAb6iP4Vhhg30xA+bKicFJq+0gDPRYJyqV8XY=";
   };
 
   nativeBuildInputs = [
     cmake
     pkg-config
+    git
```
Comment (eclairevoyant) [inline pkgs/applications/audio/clementine/default.nix:67]:
```suggestion
```
does not appear to be needed.

File: pkgs/applications/audio/clementine/default.nix [hunk 5]
```diff
@@ -39,6 +39,7 @@
 , elfutils
 , libsepol
 , orc
+, git
```
Comment (eclairevoyant) [inline pkgs/applications/audio/clementine/default.nix:42]:
```suggestion
```

**Review by eclairevoyant — COMMENTED** [review]
Please squash the commits and update the commit message + PR title to reflect the current version that you're setting here

Also, CI flagged some indentation issues: https://github.com/NixOS/nixpkgs/actions/runs/9043450163/job/24851098116?pr=310821

File: pkgs/applications/audio/clementine/default.nix [hunk 6]
```diff
@@ -118,6 +121,7 @@ in mkDerivation {
   '';
 
   cmakeFlags = [
+	  "-DFORCE_GIT_REVISION=1.3.1"
```
Comment (eclairevoyant) [inline pkgs/applications/audio/clementine/default.nix:124]:
```suggestion
    "-DFORCE_GIT_REVISION=1.3.1"
```

File: pkgs/applications/audio/clementine/default.nix (hunk 2 above)
Comment (pinpox) [reply pkgs/applications/audio/clementine/default.nix]: Updated to last commit

**Comment by qbit** [comment]
Result of `nixpkgs-review pr 310821` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clementine</li>
  </ul>
</details>

**Review by qbit — APPROVED** [review]

**Pull request merged**